crates/ruff/src/settings/types.rs

```diff
@@ -194,7 +194,7 @@ pub struct PerFileIgnore {
 
 impl PerFileIgnore {
     pub fn new(pattern: String, prefixes: &[RuleSelector], project_root: Option<&Path>) -> Self {
-        let rules: RuleSet = prefixes.iter().flat_map(IntoIterator::into_iter).collect();
+        let rules: RuleSet = prefixes.iter().flat_map(RuleSelector::all_rules).collect();
```

Comment:
This is okay (not to use preview) because this is an ignore, so it's okay if it includes rules that will never be enabled. But we should comment as such.